extensions/annotations/src/main/java/io/opentelemetry/extension/annotations/SpanAttribute.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation marks that a parameter of a method annotated by the {@link WithSpan} annotation
+ * should be added as an attribute to the newly created {@link io.opentelemetry.api.trace.Span}.
+ * Using this annotation is equivalent to calling {@code Span.currentSpan().setAttribute(...)}
+ * within the body of the method.
+ *
+ * <p>Application developers can use this annotation to signal OpenTelemetry auto-instrumentation
+ * that a new span attribute should be added to a span created when the parent method is executed.
+ *
+ * <p>If you are a library developer, then probably you should NOT use this annotation, because it
+ * is non-functional without the OpenTelemetry auto-instrumentation agent, or some other annotation
+ * processor.
+ *
+ * @see <a href="https://github.com/open-telemetry/opentelemetry-java-instrumentation">OpenTelemetry
+ *     OpenTelemetry Instrumentation for Java</a>
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface SpanAttribute {
```

Comment:
We need a `@since 1.2.0` on this class javadoc as well, if we're going to add it.

Reply:
@HaloFour Once you add this, we'll merge :)

Reply:
Sorry about that, been a bit swamped recently.  Speaking of, with this merged is there an expected turnaround time for instrumentation to match this annotation?  I've been working on some changes to my WIP PR to try to address some of the concerns brought up here but it has some more to go and I've not had a lot of opportunity to look at it this week.

Reply:
> Speaking of, with this merged is there an expected turnaround time for instrumentation to match this annotation?

that's a good thought. I hadn't thought about it, but it would be ideal for this annotation to land in the same release cycle as the support for it in the java agent to avoid user confusion (this repo usually releases first week of the month, and instrumentation repo usually releases the week after that).

makes sense to hold off on merging this PR for now then?

Reply:
I'm happy to hold off until after the release. Of course, I'd be happy to wait even longer. ;)

Reply:
@trask 

> makes sense to hold off on merging this PR for now then?

That works for me.

@jkwatson 

>  Of course, I'd be happy to wait even longer. ;)

You should see the projects I've been working on lately.  Spring WebFlux + Swagger/OpenAPI.  I think we have twice as many annotations as we have code.  😁

Reply:
Swagger is the worst. It's the example I always cite about annotation abuse.